lib/shelley/src/Cardano/Wallet/Shelley/Network.hs

```diff
@@ -328,6 +349,8 @@ withNetworkLayer tr np addrInfo versionData action = do
     connectNodeTipClient
         :: HasCallStack
         => RetryHandlers
+        -> TVar IO (Map W.ChimericAccount (Maybe W.Coin))
```

Comment:
What about using `0` as a default value instead of `Nothing`. We do already default to `0` when it comes to the reward balance and it is quite plausible / will not cause any particular harm to report a balance as 0 until we fetch it.

Reply:
I wanted to keep "we already default to 0" separate from the core caching logic.

With the slight refactoring, I made new entries get added to a separate `Set`, instead of `Nothing` values inside the `Map`, but the "query" still returns `Maybe W.Coin` like here, requiring an explicit `fromMaybe`.